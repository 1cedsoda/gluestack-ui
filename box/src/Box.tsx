import React, { forwardRef } from 'react';
import type { IBoxProps } from './types';

export function Box<T>(StyledBox: React.ComponentType<T>) {
  return forwardRef(({ children, ...props }: T & IBoxProps, ref: any) => {
    return (
      <StyledBox ref={ref} {...(props as T)}>
        {children}
      </StyledBox>
    );
  });
}
