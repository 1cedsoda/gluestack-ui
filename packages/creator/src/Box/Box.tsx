import React, { forwardRef } from 'react';

const Box = (StyledBox: any) =>
  forwardRef(({ children, ...props }: any, ref: any) => {
    return (
      <StyledBox ref={ref} {...props}>
        {children}
      </StyledBox>
    );
  });

export default Box;
