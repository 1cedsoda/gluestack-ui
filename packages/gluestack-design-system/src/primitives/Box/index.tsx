import React, { memo } from 'react';
import { Box as GlueStackBox } from '@gluestack/ui';

export const Box = memo(({ ...props }) => {
  return <GlueStackBox {...props} />;
});
