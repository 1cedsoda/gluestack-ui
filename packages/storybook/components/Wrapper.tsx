// import React from 'react';
import { GluestackUIProvider, Center, config } from '@gluestack/ui';
// import { StyledProvider } from 'dank-style';
// import { set, get } from '@gluestack/color-mode';
// import { useDarkMode } from 'storybook-dark-mode';
import React from 'react';

// window['setTheme'] = set;
// window['getTheme'] = get;
const Wrapper = ({ children }: any) => {
  // let value = useDarkMode();
  // set(value ? 'dark' : 'light');

  // console.log('config', config);

  return (
    <GluestackUIProvider config={config}>
      <Center>{children}</Center>
    </GluestackUIProvider>
  );
};

Wrapper.displayName = 'GluestackUIProvider';

export default Wrapper;
