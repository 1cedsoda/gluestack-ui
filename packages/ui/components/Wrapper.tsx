// import React from 'react';
import { Provider, Center } from '@gluestack/ui';
import { set, get } from '@gluestack/color-mode';
import { useDarkMode } from 'storybook-dark-mode';
import React from 'react';

export default ({ children }: any) => {
  let value = useDarkMode();
  set(value ? 'dark' : 'light');
  return (
    <Provider>
      <Center>{children}</Center>
    </Provider>
  );
};
