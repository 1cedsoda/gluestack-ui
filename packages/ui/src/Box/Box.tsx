import React, { forwardRef } from "react";
import { wrapStringChild } from "../utils/wrapStringChild";
import { UIContext } from "../UIProvider";

const Box = ({ children, ...props }: any, ref: any) => {
  const { StyledBox } = React.useContext(UIContext);
  return (
    <StyledBox ref={ref} {...props}>
      {({ resolveContextChildrenStyle }: any) => {
        return wrapStringChild(children, resolveContextChildrenStyle);
      }}
    </StyledBox>
  );
};

export default forwardRef(Box);
