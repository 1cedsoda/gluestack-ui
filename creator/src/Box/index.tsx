import Box from './Box';

export const createBox = ({ StyledBox }: any) => {
  const BoxTemp = Box(StyledBox) as any;
  return BoxTemp;
};
