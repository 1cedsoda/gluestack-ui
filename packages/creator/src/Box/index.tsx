import type React from 'react';
import { Box as BoxMain } from './Box';

export function createBox<T>({
  StyledBox,
}: {
  StyledBox: React.ComponentType<T>;
}) {
  const Box = BoxMain(StyledBox);
  Box.displayName = 'Box';
  return Box;
}
