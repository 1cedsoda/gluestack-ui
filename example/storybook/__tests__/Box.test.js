import React from 'react';
import renderer from 'react-test-renderer';
// import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import { render } from '@testing-library/react-native';

import { BoxStory } from '../src/components/Layout/Box/Box';
import Wrapper from '../src/components/Wrapper';

describe('Box component', () => {
  it('matches snapshot', () => {
    //@ts-ignore
    const tree = renderer
      .create(
        <Wrapper>
          <BoxStory />
        </Wrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
