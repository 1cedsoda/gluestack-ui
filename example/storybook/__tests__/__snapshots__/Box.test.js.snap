// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1yqt7l4-1q64x7k"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 yljpgc-5lm9f0"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1yqt7l4-1q64x7k"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 yljpgc-5lm9f0"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="896xuz-1q64x7k aj8jl8-2q7xs9"
          dir={null}
        >
          <div
            className="css-text-1rynq56"
            data-style="1milmwi-1k7pv9v 1milmwi-1j1xgf6 cu5vht-1a3b7b8"
            dir="auto"
          >
            BOX
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
